---
title: "Reads QC"
author: "LÃ©opold Guyot"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Packages Import 
```{r, message=FALSE, echo = FALSE}
library(ShortRead)
library(ggplot2)
```

## Functions

```{r }
mean_quality_density <- function(filename){
  # This function produces a plot of the distribution of the means of the quality
  # for each read in a file
  gc()
  print(filename)
  quality_matrix <- as(quality(readFastq(filename)), "matrix")
  means <- as.data.frame(apply(quality_matrix, 1, mean))
  names(means)[1] <- "mean"
  ggplot(means, aes(x = mean, ..scaled..))+geom_density()#change axis name
}
```

## Plot quality for each sample

You can also embed plots, for example:

```{r}
files <- list.files(path = "data/mirnaseq_data", full.names = TRUE)
# Only one sample is treated here since it is quite computationally heavy
mean_quality_density(files[1])
```

Make a plot for the mean value for each position (start and end)
Then compare with or without QC

